newIDE/app/src/InstancesEditor/SelectionRectangle.js

```diff
@@ -82,28 +155,55 @@ export default class SelectionRectangle {
     this.selectionRectangleEnd = { x, y };
   };
 
-  updateSelectionRectangle = (lastX: number, lastY: number) => {
+  updateSelectionRectangle = (
+    lastX: number,
+    lastY: number
+  ): Array<gdInitialInstance> => {
     if (!this.selectionRectangleStart)
       this.selectionRectangleStart = { x: lastX, y: lastY };
 
     this.selectionRectangleEnd = { x: lastX, y: lastY };
+    this._temporaryInstancesInSelectionRectangle.length = 0;
+    const normalizedSelectionRectangle = getRectangleNormalizedBoundaries({
+      startX: this.selectionRectangleStart.x,
+      startY: this.selectionRectangleStart.y,
+      endX: this.selectionRectangleEnd.x,
+      endY: this.selectionRectangleEnd.y,
+    });
+    this.temporarySelectionRectangleStart = {
```

Comment:
Might be slightly more efficient to just assign `this.selectionRectangleStart.x = ...` and same for y.
Avoid recreating a new object at every update.